app/main.py

```diff
@@ -2,5 +2,21 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    parameters = {}
+
+    def collector(*args) -> None:
+        if args not in parameters:
+            parameters[args] = func(*args)
+            print("Calculating new result ")
+            return func(*args)
```

Comment:
The `collector` function calls `func(*args)` twice when the result is not cached. This is redundant and inefficient. You should remove the second call to `func(*args)` on line 11 and instead return the result stored in `parameters[args]`.